devtools/onion.c

```diff
@@ -172,6 +180,19 @@ static void opt_show_ad(char buf[OPT_SHOW_LEN], const u8 *assocdata)
 	hex_encode(assocdata, ASSOC_DATA_SIZE, buf, OPT_SHOW_LEN);
 }
 
+static char *opt_set_node_id(const char *arg, struct node_id *node_id)
+{
+	node_id_from_hexstr(arg, strlen(arg), node_id);
+	return NULL;
+}
+
+static void opt_show_node_id(char buf[OPT_SHOW_LEN], const struct node_id *node_id)
+{
+	char *hex = node_id_to_hexstr(NULL, node_id);
+	assert(strlen(hex) < OPT_SHOW_LEN);
+	strncpy(buf, hex, OPT_SHOW_LEN);
+}
```

Comment:
Maybe drop the opt_show_node_id: it's used to print the default value in --help, but that's not very useful here (the key 00000000... is not very explanatory!)